File: solutions/fortran/reverse-string/1/reverse_string.f90
module reverse_string
  implicit none
contains

  function reverse(input) result(reversed)
    character(*), intent(in) :: input
    character(len=len(input)+1) :: reversed
    integer :: i
    reversed=""
    do i=1,len(input)
      reversed = reversed(1:i)//input(len(input)-i+1:len(input)-i+1)
    end do
    reversed=adjustr(reversed)
    reversed=adjustl(reversed)
  end function

end module

File: solutions/go/reverse-string/1/reverse_string.go
package reverse

func Reverse(input string) string {
    runes:= []rune(input)
    length:=len(runes)
    text:=""
	for i := 0; i < length; i++ {
		text = text + (string(runes[length-i-1]))
	}
    return text
}
